src/Tickets/Custom_Tables/V1/WP_Query/WP_Meta_Query_Modifier.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * Service for manipulating Meta Queries for Event Tickets + Custom Tables.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Tickets\Custom_Tables\V1\WP_Query;
+ */
+
+namespace TEC\Tickets\Custom_Tables\V1\WP_Query;
+
+use TEC\Events\Custom_Tables\V1\Models\Occurrence;
+use TEC\Events_Pro\Custom_Tables\V1\Models\Provisional_Post;
+use Tribe__Tickets__Attendee_Repository;
+use Tribe__Tickets__Ticket_Repository;
+use WP_Query;
+
+/**
+ * Service for manipulating Meta Queries for Event Tickets + Custom Tables.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Tickets\Custom_Tables\V1\WP_Query;
+ */
+class WP_Meta_Query_Modifier {
+
+	/**
+	 * A reference to the Attendee Repository.
+	 *
+	 * @since TBD
+	 *
+	 * @var Tribe__Tickets__Attendee_Repository
+	 */
+	protected $attendee_repository;
+
+	/**
+	 * A reference to the Ticket Repository.
+	 *
+	 * @since TBD
+	 *
+	 * @var Tribe__Tickets__Ticket_Repository
+	 */
+	protected $ticket_repository;
+
+	/**
+	 * A list of meta keys to watch out for.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string>
+	 */
+	protected $meta_keys = [];
+
+	/**
+	 * The Provisional Post object.
+	 *
+	 * @since TBD
+	 *
+	 * @var Provisional_Post
+	 */
+	protected $provisional_post;
+
+	/**
+	 * Post types we are watching for.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string>
+	 */
+	protected $post_types = [];
+
+	/**
+	 * Ticket Meta Query Modifier constructor.
+	 *
+	 * @since TBD
+	 *
+	 * @param Tribe__Tickets__Attendee_Repository $attendee_repository
+	 * @param Tribe__Tickets__Ticket_Repository   $ticket_repository
+	 * @param Provisional_Post                    $provisional_post
+	 */
+	public function __construct(
+		Tribe__Tickets__Attendee_Repository $attendee_repository,
```

Comment:
[In the provider](https://github.com/the-events-calendar/event-tickets/pull/2529/files#diff-0ad318c81052f97ad003bd713b8295ff1ec0673e3a574c11cebcec75e509d085R49) you're setting this as a singleton object, its dependencies will be injected once, when the instance is first built.
While it's fine to require an instance of the Attendee Repository and Ticket Repository to get  the `post_types` out, as it's done in the body of the `__construct` method some lines later, the method should not keep references to the particular instances of the repository in the `attendee_repository` and `ticket_repository` properties: the repository instances are built and destroyed for each query and someone could be tempted to think those properties reference some singleton instances of the repositories.
The repositories are not singleton by design, e.g. how the Ticket Repository is bound as not singleton [here](https://github.com/the-events-calendar/event-tickets/blob/8d569126e399f71000273d752505cf506d07f807/src/Tribe/Service_Providers/ORM.php#L25).

Reply:
Good catch, I actually meant to refactor those out after a cleanup iteration. Whoops.